### PR TITLE
Xray: added xray_get_rcv_event_sock. return sock_fd that can be sent …

### PR DESCRIPTION
**Pull request by imenashe-dn**
Xray: added xray_get_rcv_event_sock. return sock_fd that can be sent to poll/select

### CODE REVIEW

File: src/xray.cpp [hunk 1]
```diff
@@ -1313,6 +1323,19 @@ int xray_dump(const char *path, char **out_str)
     return -1;
 }
 
+int
+xray_get_rcv_event_sock(int *res)
```
Comment (grishaf) [inline src/xray.cpp]: I want to add support for XrayServer(cluster). XrayNode will open two sockets - one local uds (as today) and one to XrayServer. can you make this function api:
`int xray_get_rcv_event_sock(int *rx_sockets, int *n_sockets)`

**Review by grishaf — CHANGES_REQUESTED** [review]
please add tests in order this functionalliy will not be broken. run `make test` or `make test-in-docker` to run the tests

**Comment by kfir-drivenets** [comment]
what about time related triggers? seems to me like waiting only on the file descriptors has the potential to break the rate computation feature

**Comment by grishaf** [comment]
@kfir-drivenets you are right, with list it can be fixed easily. BTW, what is better list or mask?

**Comment by kfir-drivenets** [comment]
I think that its similar to select vs epoll.
Select uses a mask with 1024 bits.
Epoll uses a list.
Epoll is much more scalable for large number of file descriptors.
I wouldn't go with the mask approach.

**Pull request closed without merging**